qcodes/dataset/measurements.py

```diff
@@ -862,9 +870,11 @@ def add_after_run(self, func: Callable, args: tuple) -> None:
 
         self.exitactions.append((func, args))
 
-    def add_subscriber(self,
+        return self
+
+    def add_subscriber(self : T,
```

Comment:
could you add (or modify existing) a test that uses (all at once) these methods (for no good reason) with the fluidity, just so that this behavior of the methods is captured?

Reply:
On it.